### ISSUE TITLE
The `decay_params` parameter is invalid

### ISSUE THREAD

**Issue by kuviki** [SakanaAI/continuous-thought-machines#5]
Hello, while reviewing the code, I noticed that `torch.clamp` is used in the parameters `decay_params_action` and `decay_params_out`. During model training, if the parameter value becomes less than 0, it is truncated to a constant 0, causing the gradient of the parameters to always be 0. For example, the following line of code, as well as other similar instances in the code:

https://github.com/SakanaAI/continuous-thought-machines/blob/2a1b93acfb95d3ce67057d24fdb03cc7ee409491/models/ctm.py#L503

To implement this correctly, it can be modified to:
```python
self.decay_params_action.data = torch.clamp(self.decay_params_action, 0, 15)
self.decay_params_out.data = torch.clamp(self.decay_params_out, 0, 15)
r_action, r_out = torch.exp(-self.decay_params_action).unsqueeze(0).repeat(B, 1), torch.exp(-self.decay_params_out).unsqueeze(0).repeat(B, 1)
```
